src/compiler/checker.ts

```diff
@@ -5299,7 +5299,7 @@ namespace ts {
          * Check if a Type was written as a tuple type literal.
          * Prefer using isTupleLikeType() unless the use of `elementTypes` is required.
          */
-        function isTupleType(type: Type): boolean {
+        function isTupleType(type: Type): type is TupleType {
             return (type.flags & TypeFlags.Tuple) && !!(<TupleType>type).elementTypes;

```

Comment:
Wait, I think the idea was that @caitp added this so that we could report better errors when a tuple type literal is available (#2238).